nni/common/serializer.py

```diff
@@ -308,40 +511,47 @@ def _import_cls_or_func_from_hybrid_name(s: str) -> Any:
 
 
 def _json_tricks_func_or_cls_encode(cls_or_func: Any, primitives: bool = False, pickle_size_limit: int = 4096) -> str:
-    if not isinstance(cls_or_func, type) and not callable(cls_or_func):
+    if not isinstance(cls_or_func, type) and not _is_function(cls_or_func):
         # not a function or class, continue
         return cls_or_func
 
     return {
-        '__nni_type__': _get_hybrid_cls_or_func_name(cls_or_func, pickle_size_limit)
+        '__nni_type__': get_hybrid_cls_or_func_name(cls_or_func, pickle_size_limit)
     }
 
 
 def _json_tricks_func_or_cls_decode(s: Dict[str, Any]) -> Any:
     if isinstance(s, dict) and '__nni_type__' in s:
         s = s['__nni_type__']
-        return _import_cls_or_func_from_hybrid_name(s)
+        return import_cls_or_func_from_hybrid_name(s)
     return s
 
 
 def _json_tricks_serializable_object_encode(obj: Any, primitives: bool = False, use_trace: bool = True) -> Dict[str, Any]:
     # Encodes a serializable object instance to json.
-    # If primitives, the representation is simplified and cannot be recovered!
 
     # do nothing to instance that is not a serializable object and do not use trace
-    if not use_trace or not isinstance(obj, SerializableObject):
+    if not use_trace or not is_traceable(obj):
         return obj
 
-    return obj.__json_encode__()
+    if isinstance(obj.trace_symbol, property):
+        # commonly made mistake when users forget to call the traced function/class.
+        warnings.warn(f'The symbol of {obj} is found to be a property. Did you forget to create the instance with ``xx(...)``?')
```

Comment:
this is only a warning, not an error?

Reply:
We can't be sure. In rare cases, they might actually want this as expected behavior.